### ISSUE TITLE
Search function not working properly

### ISSUE THREAD

**Issue by madmaharaja** [YOURLS/YOURLS#3657]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/YOURLS/.github/blob/main/CODE_OF_CONDUCT.md).

### Submission validity

- [X] This is not a personal support request, that should be posted on [discussions community](https://github.com/YOURLS/YOURLS/discussions).
- [X] I checked [current issues](https://github.com/YOURLS/YOURLS/issues) and this request isn't a duplicate of an existing issue, opened or closed.

### Self troubleshooting

- [X] I've read the [documentation](https://yourls.org/) and made sure to follow it.
- [X] I've read the [troubleshooting first steps](https://docs.yourls.org/guide/troubleshooting/first-steps.html) and [frequent issues](https://docs.yourls.org/guide/troubleshooting/frequent-issues.html) guides.

### Version

1.9.2

### Description

The search function is giving me trouble. If I search for any given keyword that is either in the short link, in the original URL or in the title then in some cases no results are returned and sometimes a few (but not all existing) results are returned.

Here is a random example that I created to demonstrate the problem:
I created multiple entries with the keyword "abcdefg" in different fields:
![grafik](https://github.com/YOURLS/YOURLS/assets/3133710/7a1305cd-e6cf-4faa-8018-3ce449a17587)

When I search for "abcdefg" no results are found:
![grafik](https://github.com/YOURLS/YOURLS/assets/3133710/1137f95d-3b60-4d62-98e4-e3f6ab39c04e)

But as mentioned -- there are other keywords where a few results are found by the search but not all of the existing instances.

I have tested it by deactivating all plugins -- unfortunately with the same result.
Does anyone have an idea what the problem could be? Thanks for any suggestions!

**Comment by dgw** [comment]
I have established that this feature is case-sensitive on my live instance of 1.9.2, though I don't believe this explains any of the missing example entries shown in the top screenshot.

However, the default search filter is `more than 0 clicks`. If I add a new link and don't click it, searching for that link by keyword only without modifying any other form fields does not in fact return it in the list.

Manually putting `-1` in the click-count field returns results including the 0-click link as expected, and a form with the click-count field empty for further searches. However, submitting this new form with no value in the click-count makes it default to 0, filtering out the new link again.

Based on this, I will update the issue title with what I believe to be a concise description of the true bug.

**Comment by dgw** [comment]
It is possible to reproduce the behavior by visiting `YOURLS_SITE . '/admin/?click_limit='` only. None of the other URL parameters are necessary to observe links with 0 clicks being filtered out of the results.

**Comment by madmaharaja** [comment]
> I have established that this feature is case-sensitive on my live instance of 1.9.2, though I don't believe this explains any of the missing example entries shown in the top screenshot.
> 
> However, the default search filter is `more than 0 clicks`. If I add a new link and don't click it, searching for that link by keyword only without modifying any other form fields does not in fact return it in the list.
> 
> Manually putting `-1` in the click-count field returns results including the 0-click link as expected, and a form with the click-count field empty for further searches. However, submitting this new form with no value in the click-count makes it default to 0, filtering out the new link again.
> 
> Based on this, I will update the issue title with what I believe to be a concise description of the true bug.

Thanks for your quick reply! 
I can confirm: When I manually enter "-1" in the click-count field, all the entries show up correctly in the search:
![grafik](https://github.com/YOURLS/YOURLS/assets/3133710/802e0773-e1df-44d5-a481-478c8c51026f)

So, is there a way to change the default click limitation in the config.php?

**Comment by dgw** [comment]
The default is supposed to be `''` (none). I did a little bit of poking around the code earlier to try and figure out what's going on, and didn't find anything yet. All the relevant code I examined appears to correctly interpret `''` as "no limit", so I'm unsure where the empty parameter is somehow turning into a `0` equivalent.

This isn't something that should need configuring. It is indeed a bug that empty == 0; I just don't know where the bug is.

(The other people on YOURLS' maintenance team are much better at PHP than I am, so when they get a chance to look into this issue we'll surely have a better idea of what's going on.)

**Comment by sascha-hendel** [comment]
Noticed this issue, too.
I have a fresh installation of YOURLS done with composer.

Reason is a bug in file `https://github.com/YOURLS/YOURLS/blob/master/includes/Views/AdminParams.php` in function `get_click_limit()` (line 222-227). On inspecting the code I saw that my file is different than the master in github!  The problem is already solved (use of empty() instead of isset() ) by this pull request: https://github.com/YOURLS/YOURLS/pull/3589 from July

I'm wonder, why my composer installation (installed 2 days ago) still contains this bug? The admin interface displays: Powered by [YOURLS](http://yourls.org/) v 1.9.2

**Comment by dgw** [comment]
> I'm wonder, why my composer installation (installed 2 days ago) still contains this bug? The admin interface displays: Powered by YOURLS v 1.9.2

1.9.2 released (was tagged) in March, and the patch was merged in July.

**Comment by ThanhPhongPhan** [comment]
***@***.***

Vào 23:12, Th 3, 19 thg 12, 2023 dgw ***@***.***> đã viết:

**Comment by sascha-hendel** [comment]
ok, but anyway this bug report can be closed now ;)

**Comment by dgw** [comment]
True! And it being fixed in #3589 explains why I couldn't find where the value was mishandled earlier in this thread. :)
